testutils/infra/container_manager/docker_compose_manager.py

```diff
@@ -336,6 +336,8 @@ def setup(self):
 class DockerComposeMonitorCommercialSetup(DockerComposeNamespace):
     def __init__(self, name, num_clients=0):
         self.num_clients = num_clients
+        # Lock to avoid concurrent tests using SMTP mock (port 1025)
+        self.smtp_mock_lock = filelock.FileLock("smtp_mock_lock")
```

Comment:
If all monitoring tests are going to use this, it's going to be slow because of lacking parallelism, and will get slower for every test we add. A couple of suggestions:

* Move the smtp part to a separate setup class and use it only in the tests that really require smtp.
  _-or-_
* Don't use a publicly exposed port, but instead set the hostname of the server by overriding the `WORKFLOWS_SMTP_HOST` variable when launching the environment. This allows it to use the internal server in its Docker network instead of a public address. This is the best solution, IMHO, since they can run in parallel like any other test.

Reply:
The second alternative sounds much better, of course.

I suggest we merge this as-is now (pipelines are failing as we speak), and I start working on your second approach straight away.

Reply:
:+1:

Reply:
Actually, I am running the tests locally without exposing the port out of the Docker network and it "just works". I don't think this was needed in the first place... I am doing more local tests to be sure

Reply:
Nah, I was wrong. The tests need a bit of rework. Merging as agreed and coming up with a better fix soon :+1: